R/utils.R

```diff
@@ -259,85 +259,560 @@ rowcol_to_dim <- function(row, col) {
+#'  data with column names, and without row names. as the code is more clean.
+#'
+#' In the `add_data()` / `add_font()` example, if writing the data with row names
+#'
+#'
+#' ```r
+#' dims_row_names <- wb_dims(x = mtcars, row_names = TRUE, cols = 0)
+#' # add data to an object with row names
+#' wb <- wb_workbook()
+#' wb$add_worksheet("test")
+#' full_mtcars_dims <-
+#' wb$add_data(x = mtcars, dims = wb_dims(x = mtcars, row_names = TRUE), row_names = TRUE)
+#' # Style row names of an object (many options)
+#' # The programmatic way to access row names only with `x` is
+#' dims_row_names <- wb_dims(x = mtcars, row_names = TRUE, col_names = FALSE, cols = 0, from_col = 0)
+#' # In this case, it's much better to use a simpler alternative without using `x`
+#' dims_row_names <- wb_dims(cols = "A", from_row = 2)
+#' dims_row_names <- wb_dims(2:33, 1) # or dims <- "A2:A33"
+#' dims_row_names <- "A2:A33" # or simply "A2"
+#' wb$add_font(dims = dims_row_names, bold = TRUE)
+#' # the following would work too, but `wb_dims()` may be longer to write, but easier to read after, as
+#' # it can make it clear which object is affected
+#' wb$add_font(dims = dims_row_names, bold = TRUE)
+#'
+#' ```
+#'
+#' @param ... construct dims arguments, from rows/cols vectors or objects that
+#'   can be coerced to data frame
+#' @return A `dims` string
+#' @export
 #' @examples
-#' # either vectors
+#' wb_dims(x = mtcars, row_names = TRUE, col_names = FALSE, cols = 0)
+#' # Provide coordinates
+#' wb_dims()
+#' wb_dims(1, 4)
+#' wb_dims(rows = 1, cols = 4)
+#' wb_dims(from_row = 4)
+#' wb_dims(from_col = 2)
+#' wb_dims(1:4, 6:9, from_row = 5)
+#' # Provide vectors
+#' wb_dims(1:10, c("A", "B", "C"))
 #' wb_dims(rows = 1:10, cols = 1:10)
+#' # provide `from_col` / `from_row`
+#' wb_dims(rows = 1:10, cols = c("A", "B", "C"), from_row = 2)
+#' wb_dims(rows = 1:10, cols = 1:10, from_col = 2)
 #' # or objects
-#' wb_dims(mtcars)
-#' @export
+#'
+#' wb_dims(x = mtcars)
+#' # column names of an object (with the special `rows = 0`)
+#' wb_dims(x = mtcars, rows = 0)
+#' # usually, it's better
+#' # dims of all the data of mtcars.
+#' wb_dims(x = mtcars, col_names = FALSE)
+#'
+#' # dims of the column names of an object
+#' wb_dims(x = mtcars, rows = 0, col_names = TRUE)
+#'
+#' ## add formatting to column names with the help of `wb_dims()` ====
+#' wb <- wb_workbook()
+#' wb$add_worksheet("test")
+#' wb$add_data(x = mtcars, dims = wb_dims(x = mtcars))
+#' # Style col names of an object to bold (many options)
+#' \dontrun{
+#' wb <- wb_workbook()
+#' # Supplying dims using x
+#' dims_column_names <- wb_dims(x = mtcars, rows = 0)
+#' wb$add_font(dims = dims_column_names, bold = TRUE)
+#'
+#' # Finally, to add styling to column "cyl" (the 4th column)
+#' # there are many options, but here is the preferred one
+#' # if you know the column index, wb_dims(x = mtcars, cols = 4) also works.
+#' dims_cyl <- wb_dims(x = mtcars, cols = "cyl")
```

Comment:
Would the following be more informative
          "Supplying multiple column names is not supported by the `wb_dims()` helper, ",
          "use the `cols` with a range instead of `x` column names.",
          "\n Use a single `cols` at a time with `wb_dims()`"

Reply:
Thanks, I guess we could see if we want to extend it in the future. After all it would simply be a tiny loop over select arguments and `paste(dims, collapse = ";")` at the end. But sometimes you put a lot of effort into something and in the end it will only be used by you and only while you are developing the feature. We can wait until feature requests appear (in all honesty I don't want to review `wb_dims()` related stuff in the very near future :wink: ).